test/serializers/root_test.rb

```diff
@@ -3,18 +3,23 @@
 module ActiveModel
   class Serializer
     class RootTest < Minitest::Test
+      class VirtualValue < ::Model
+      end
+      class VirtualValueSerializer < ActiveModel::Serializer
+        attributes :id
+      end
       def setup
         @virtual_value = VirtualValue.new(id: 1)
       end
 
       def test_overwrite_root
-        serializer = VirtualValueSerializer.new(@virtual_value, { root: 'smth' })
+        serializer = VirtualValueSerializer.new(@virtual_value, root: 'smth')
         assert_equal('smth', serializer.json_key)
       end
 
       def test_underscore_in_root
         serializer = VirtualValueSerializer.new(@virtual_value)
-        assert_equal('virtual_value', serializer.json_key)
+        assert_equal('active_model/serializer/root_test/virtual_value', serializer.json_key)
```

Comment:
More precisely, `VirtualValue` was defined in the global namespace before, so its `json_key` was not namespaced.